File: landscape-webui/src/lib/wifi.ts
export class WifiServiceConfig {
  iface_name: string;
  enable: boolean;
  config: string;

  constructor(obj?: { iface_name: string; enable?: boolean; config?: string }) {
    this.iface_name = obj?.iface_name ?? "";
    this.enable = obj?.enable ?? true;
    this.config = obj?.config ?? "";
  }
}

File: landscape-common/src/config/wifi.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

use crate::store::storev2::LandscapeStore;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export, export_to = "common/config.d.ts")]
pub struct WifiServiceConfig {
    pub iface_name: String,
    pub enable: bool,
    /// hostapd config file
    pub config: String,
}

impl LandscapeStore for WifiServiceConfig {
    fn get_store_key(&self) -> String {
        self.iface_name.clone()
    }
}
